### ISSUE TITLE
Allow creating model and saving file in one request 

### ISSUE THREAD

**Issue by abzainuddin** [CodeYellowBV/django-binder#45]
Now you have to:

1. save the model
2. save the file

Maybe some special sauce for multiput?

**Comment by stefanmajoor** [comment]
Maybe integrate the ImageView from HN directly too?

**Comment by mmoreaux** [comment]
@abzainuddin But file uploads require a `multipart/form-data` body, rather than a plain json body.

I can see two possible ways to achieve this:
 - Post the image base64-encoded as part of the json
 - Post both the image and the json body in a `multipart/form-data` body

**Comment by abzainuddin** [comment]
POSTing base64 might be slow, especially for bigger files / mobile devices. I would prefer option 2.

**Comment by mmoreaux** [comment]
Would option 2 be reasonable to implement in the FE?

**Comment by SpaceK33z** [comment]
@mmoreaux yes (Burhan asked me to approve this). So in the form data there would be a `file` field and a `data` field, where the `file` field contains the blob and the `data` field stringified JSON, right?

**Comment by mmoreaux** [comment]
@SpaceK33z Something like that, yes. Would that be OK?

The details will probably differ a bit, e.g. to accommodate multiple images on a single entity, or on a single multiput. But that mainly involves having unique names for the file parts, and attaching them to the proper file field.

Maybe like this:

`POST /api/foo/`

form-data `data`: text

```
{
	"name": "Foo",
	"barness:" 5,
	"cover_image": "form:cover_image1",
	"other_image": "form:c440aa6"
}
```

form-data `file_cover_image1`: file

form-data `file_c440aa6`: file

(having the `form:` prefix allows us to add `base64:` and other prefixes in the future)

**Comment by SpaceK33z** [comment]
Yes, that would be okay.
